File: Dsa-day201.java
class Solution {
    public boolean isTrionic(int[] nums) {
        int n = nums.length;
        
        for (int p = 1; p < n - 2; p++) {
            for (int q = p + 1; q < n - 1; q++) {
                boolean valid = true;
                
                for (int i = 0; i < p; i++) {
                    if (nums[i] >= nums[i + 1]) {
                        valid = false;
                        break;
                    }
                }
                
                if (valid) {
                    for (int i = p; i < q; i++) {
                        if (nums[i] <= nums[i + 1]) {
                            valid = false;
                            break;
                        }
                    }
                }
                
                if (valid) {
                    for (int i = q; i < n - 1; i++) {
                        if (nums[i] >= nums[i + 1]) {
                            valid = false;
                            break;
                        }
                    }
                }
                
                if (valid) {
                    return true;
                }
            }
        }
        
        return false;
    }
}

File: Dsa-day201.py
class Solution:
    def isTrionic(self, nums: List[int]) -> bool:
        n = len(nums)
        
        for p in range(1, n - 2):
            for q in range(p + 1, n - 1):
                valid = True
                
                for i in range(p):
                    if nums[i] >= nums[i + 1]:
                        valid = False
                        break
                
                if valid:
                    for i in range(p, q):
                        if nums[i] <= nums[i + 1]:
                            valid = False
                            break
                
                if valid:
                    for i in range(q, n - 1):
                        if nums[i] >= nums[i + 1]:
                            valid = False
                            break
                
                if valid:
                    return True
        
        return False